src/vectorflow/serde.d

```diff
@@ -52,14 +52,32 @@ class Serializer {
     final T read(T)() if(isSomeString!T || isBasicType!T || isUnsigned!T)
     {
         scope(failure){ if(_f.eof) throw new EOFException("");}
-        scope(exit){ if(_f.eof) throw new EOFException("");}
+        version(Windows){} // Windows requires special treatment with EOF
+        else scope(exit){ if(_f.eof) throw new EOFException(""); }
         static if(isSomeString!T)
         {
-            ulong str_sz;
-            _f.rawRead((&str_sz)[0..1]);
-            auto str = new ubyte[str_sz.to!size_t];
-            str = _f.rawRead(str);
-            return cast(string)str;
+            // On Windows EOF is not set before you try to read past the end, so it requires some special handling
```

Comment:
Is this actually Windows specific? On Linux I get an assertion failure too, with this program:
```d
import std.file;
import std.stdio;

void main()
{
	std.file.write("test.txt", "Test\n");

	auto f = File("test.txt");
	f.readln();
	assert(f.eof);
}
```

Reply:
To be fair, I didn't do much checking - the tests failed on Windows and they didn't fail on Linux. Therefore I debugged a little and found out this. I can't remember the details anymore

Reply:
This really looks like it needs to be fixed somewhere else.